### PR TITLE
chore: provided more specific error for wrong import method

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes https://emqx.atlassian.net/browse/EMQX-12239
Release version: v/e5.7.0
image

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_auth_mnesia/src/emqx_authn_mnesia.erl [hunk 1]
```diff
@@ -527,6 +527,10 @@ find_password_hash(hash, User = #{<<"password_hash">> := PasswordHash}, _) ->
     {PasswordHash, maps:get(<<"salt">>, User, <<>>)};
 find_password_hash(plain, #{<<"password">> := Password}, Algorithm) ->
     emqx_authn_password_hashing:hash(Algorithm, Password);
+find_password_hash(hash, _User, _) ->
+    error("hash_import_required_password_hash_field");
+find_password_hash(plain, _User, _Algorithm) ->
+    error("plain_import_required_password_field");
```
Comment (zmstone) [inline apps/emqx_auth_mnesia/src/emqx_authn_mnesia.erl:533]:
```suggestion
    error("plain_import_requires_password_field");
```

File: apps/emqx_auth_mnesia/src/emqx_authn_mnesia.erl [hunk 2]
```diff
@@ -527,6 +527,10 @@ find_password_hash(hash, User = #{<<"password_hash">> := PasswordHash}, _) ->
     {PasswordHash, maps:get(<<"salt">>, User, <<>>)};
 find_password_hash(plain, #{<<"password">> := Password}, Algorithm) ->
     emqx_authn_password_hashing:hash(Algorithm, Password);
+find_password_hash(hash, _User, _) ->
+    error("hash_import_required_password_hash_field");
```
Comment (zmstone) [inline apps/emqx_auth_mnesia/src/emqx_authn_mnesia.erl:531]:
```suggestion
    error("hash_import_requires_password_hash_field");
```

File: apps/emqx_auth_mnesia/test/emqx_authn_mnesia_SUITE.erl [hunk 3]
```diff
@@ -228,6 +228,14 @@ t_import_users(_) ->
             State
         )
     ),
+    %% import plain.json with hash method
+    ?assertEqual(
+        {error, "hash_import_required_password_hash_field"},
```
Comment (zhongwencool) [inline apps/emqx_auth_mnesia/test/emqx_authn_mnesia_SUITE.erl:233]:
```suggestion
        {error, "hash_import_requires_password_hash_field"},
```

File: apps/emqx_auth_mnesia/test/emqx_authn_mnesia_SUITE.erl [hunk 4]
```diff
@@ -289,6 +297,14 @@ t_import_users_plain(_) ->
             State
         )
     ),
+    %% import hash.json with plain method
+    ?assertEqual(
+        {error, "plain_import_required_password_field"},
```
Comment (zhongwencool) [inline apps/emqx_auth_mnesia/test/emqx_authn_mnesia_SUITE.erl:302]:
```suggestion
        {error, "plain_import_requires_password_field"},
```

**Review by HJianBo — APPROVED** [review]

**Comment by emqxqa** [comment]
[Test Execution]()

**Pull request merged**